File: 1071-BinaryPrefixDivisibleBy5/1071-BinaryPrefixDivisibleBy5.java
// Last updated: 04/04/2026, 23:32:14
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> ans = new ArrayList<>(nums.length);
        int val = 0;
        for(int bit : nums) {
            val = ((val<<1)+bit) % 5;
            ans.add(val==0);
        }
        return ans;
    }
}

File: 1071-BinaryPrefixDivisibleBy5/1071-BinaryPrefixDivisibleBy5.cpp
// Last updated: 04/04/2026, 23:32:11
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int n = nums.size();
        vector<bool> ans(n);
        int val = 0;
        for(int i=0;i<n;i++) {
            val = ((val<<1)+nums[i]) % 5;
            ans[i] = (val==0);
        }
        return ans;
    }
};